modules/Bio/EnsEMBL/Variation/Publication.pm

```diff
@@ -282,4 +292,66 @@ sub variations{
    return $self->{'variants'};
 }
 
+=head2 set_variation_id_to_source
+
+  Arg [1]    : Integer $variation_id
+               String $source
+  Example    : $obj->set_variation_id_to_source(1234, 'dbSNP')
+  Description: Setter for the publication sources
+  Returntype : none
+  Exceptions : throw if variation_id is not valid
+               throw if source is not valid
+  Caller     : general
+  Status     : At Risk
+
+=cut
+
+sub set_variation_id_to_source{
+  my $self         = shift;
+  my $variation_id = shift;
+  my $source       = shift;
+
+  if($variation_id !~ /^\d+$/ || !defined($variation_id)) {
+    throw('variation_id is not valid');
+  }
+
+  if(!defined($source) || !defined($SOURCES{$source})) {
+    throw('Source is not valid');
+  }
+
+  return $self->{'variation_id_to_source'}->{$variation_id}->{$source} = 1;
```

Comment:
You can remove the return here. It is only a setter method and doesn't return anything.